src/utils/create-slots.tsx

```diff
@@ -1,104 +1,72 @@
+import {omit} from 'lodash'
 import React from 'react'
-import {useForceUpdate} from './use-force-update'
 import useLayoutEffect from './useIsomorphicLayoutEffect'
 
 /** createSlots is a factory that can create a
  *  typesafe Slots + Slot pair to use in a component definition
  *  For example: ActionList.Item uses createSlots to get a Slots wrapper
  *  + Slot component that is used by LeadingVisual, Description
+ *
+ * ⚠️ WARNING: Providing a custom `Context` in the same component as `useSlots` is called
+ * will result in an infinite render loop unless you memoize that context object!
```

Comment:
This is a frustrating gotcha - if you don't memoize the context, the component will get stuck in an infinite cycle of re-renders. This is because the `Slot` has a matching `useContext`, so an update in the context will re-render the `Slot`. When the `Slot` renders, it registers itself, re-rendering the component with the `useSlots` hook. Since the context is defined in that component, the context updates again, _ad infinitum_.

When `Slots` was a component instead of the `useSlots` hook, this wasn't as big of an issue because you could put the context provider outside of the `Slots` component in the JSX and avoid the infinite renders. That was still a gotcha though, so I don't think it's worth reverting the hook change to fix this.

We could bring back a custom `context` prop on `SlotsProvider` and avoid the infinite render bug, but it would significantly hurt performance because the `SlotsContext` could no longer be effectively memoized.

I'm open to ideas on improving this. It's not a bad idea to memoize context objects anyway, but in this case it feels like engineers will run into this problem regularly and the solution is not obvious. Is there a way we can at least catch the infinite render error and show more useful information?